third_party/3/pyspark/sql/context.pyi

```diff
@@ -62,3 +62,5 @@ class UDFRegistration:
     sqlContext = ...  # SQLContext
     def __init__(self, sqlContext: SQLContext) -> None: ...
     def register(self, name: str, f: Callable[..., Any], returnType: DataType = ...) -> None: ...
+    def registerJavaFunction(self, name: str, javaClassName: str, returnType: Optional[DataType] = ...) -> None: ...
```

Comment:
`returnType` supports both `DataType` and `str` (via [`_parse_datatype_string`](https://github.com/apache/spark/blob/c353a84d1a991797f255ec312e5935438727536c/python/pyspark/sql/udf.py#L395)) so we should use [`DataTypeOrString`](https://github.com/zero323/pyspark-stubs/blob/f17fb805b0d09f7799b124a89b39d0f93a137b1e/third_party/3/pyspark/sql/_typing.pyi#L18) here.

Reply:
For that matter, this applies to `UDFRegistration.register` as well, since it delegates return type handling to `UserDefinedFunction`. Just in case you want to tackle this as well.

Reply:
Thanks for the correctness.